Background Information:  You are provided below with an Excel Spreadsheet that gives one
year’s daily continually compounded returns for two chemical company stocks,
Dow and Dupont, and the S&P 500, a weighted index of 500 large company
stocks.

In [1]:
import numpy as np
import pandas as pd

from scipy.stats import linregress

In [2]:
df = pd.read_csv("Week-1-Practice-Quiz-Spreadsheet.csv", parse_dates=["Date"], dayfirst=True)

In [3]:
df

Date  DowChemical  Dupont  S&P500
0   2010-12-31       -0.003   0.002   0.000
1   2010-12-30        0.001   0.005  -0.002
2   2010-12-29        0.000   0.012   0.001
3   2010-12-28        0.001   0.009   0.001
4   2010-12-27       -0.002   0.002   0.001
..         ...          ...     ...     ...
247 2010-01-08        0.004  -0.022   0.003
248 2010-01-07        0.001  -0.021   0.004
249 2010-01-06        0.018   0.001   0.001
250 2010-01-05        0.044  -0.028   0.003
251 2010-01-04        0.055  -0.003   0.016

[252 rows x 4 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 252 entries, 0 to 251
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   Date         252 non-null    datetime64[ns]
 1   DowChemical  252 non-null    float64       
 2   Dupont       252 non-null    float64       
 3   S&P500       252 non-null    float64       
dtypes: datetime64[ns](1), float64(3)
memory usage: 8.0 KB


Problem Information:  Daily continuously compounded returns can be summed to
obtain returns over longer time intervals. Sum the daily returns to calculate
annual continuously compounded returns for 2010.  Give each result in percent, rounded to two
digits to the right of the decimal place – for example, 11.76%.

Solve: What is the Dow Chemical Annual return?

In [5]:
np.sum(df["DowChemical"])

0.24500000000000005

In [6]:
0.245 / 365

0.0006712328767123287

Question:  What is the correlation between daily continuously compounded returns for Dow Chemical and for the S&P 500 Index? Round your answer two digits to the right of the decimal place - for example, .84

In [7]:
df.corr()

DowChemical    Dupont    S&P500
DowChemical     1.000000  0.476948  0.785056
Dupont          0.476948  1.000000  0.568935
S&P500          0.785056  0.568935  1.000000

Question:  On what day in 2010 did Dow Chemical returns out perform S&P 500 Index returns the most?

In [8]:
df["dow>s&p"] = df["DowChemical"] - df["S&P500"]

In [9]:
df

Date  DowChemical  Dupont  S&P500  dow>s&p
0   2010-12-31       -0.003   0.002   0.000   -0.003
1   2010-12-30        0.001   0.005  -0.002    0.003
2   2010-12-29        0.000   0.012   0.001   -0.001
3   2010-12-28        0.001   0.009   0.001    0.000
4   2010-12-27       -0.002   0.002   0.001   -0.003
..         ...          ...     ...     ...      ...
247 2010-01-08        0.004  -0.022   0.003    0.001
248 2010-01-07        0.001  -0.021   0.004   -0.003
249 2010-01-06        0.018   0.001   0.001    0.017
250 2010-01-05        0.044  -0.028   0.003    0.041
251 2010-01-04        0.055  -0.003   0.016    0.039

[252 rows x 5 columns]

In [10]:
df["dow>s&p"].max()

0.051000000000000004

In [11]:
df[df["dow>s&p"] == 0.051000000000000004]

Date  DowChemical  Dupont  S&P500  dow>s&p
172 2010-04-28        0.057  -0.018   0.006    0.051

Excel Problem Type:  Using Excel “If” statements to determine how many days in 2010 Dow Chemical returns are higher than Dupont Returns.

Question:  How many days out of the 252 trading days in 2010 did Dow outperform Dupont?

In [12]:
df["dow>dupont"] = df["DowChemical"] - df["Dupont"]

In [13]:
df.head()

Date  DowChemical  Dupont  S&P500  dow>s&p  dow>dupont
0 2010-12-31       -0.003   0.002   0.000   -0.003      -0.005
1 2010-12-30        0.001   0.005  -0.002    0.003      -0.004
2 2010-12-29        0.000   0.012   0.001   -0.001      -0.012
3 2010-12-28        0.001   0.009   0.001    0.000      -0.008
4 2010-12-27       -0.002   0.002   0.001   -0.003      -0.004

In [14]:
df["flag"] = df["dow>dupont"].apply(lambda x: 1 if x > 0 else 0)

In [15]:
df

Date  DowChemical  Dupont  S&P500  dow>s&p  dow>dupont  flag
0   2010-12-31       -0.003   0.002   0.000   -0.003      -0.005     0
1   2010-12-30        0.001   0.005  -0.002    0.003      -0.004     0
2   2010-12-29        0.000   0.012   0.001   -0.001      -0.012     0
3   2010-12-28        0.001   0.009   0.001    0.000      -0.008     0
4   2010-12-27       -0.002   0.002   0.001   -0.003      -0.004     0
..         ...          ...     ...     ...      ...         ...   ...
247 2010-01-08        0.004  -0.022   0.003    0.001       0.026     1
248 2010-01-07        0.001  -0.021   0.004   -0.003       0.022     1
249 2010-01-06        0.018   0.001   0.001    0.017       0.017     1
250 2010-01-05        0.044  -0.028   0.003    0.041       0.072     1
251 2010-01-04        0.055  -0.003   0.016    0.039       0.058     1

[252 rows x 7 columns]

In [16]:
df["flag"].value_counts()

0    129
1    123
Name: flag, dtype: int64

Question:  What was the fifth-worst performing day for the S&P 500 Index in 2010?

In [17]:
df.sort_values(by="S&P500")

Date  DowChemical  Dupont  S&P500  dow>s&p  dow>dupont  flag
156 2010-05-20       -0.032  -0.059  -0.040    0.008       0.027     1
146 2010-06-04       -0.049  -0.058  -0.035   -0.014       0.009     1
166 2010-05-06       -0.063  -0.033  -0.033   -0.030      -0.030     0
129 2010-06-29       -0.055  -0.041  -0.032   -0.023      -0.014     0
229 2010-02-04       -0.036  -0.043  -0.032   -0.004       0.007     1
..         ...          ...     ...     ...      ...         ...   ...
142 2010-06-10        0.037   0.056   0.029    0.008      -0.019     0
84  2010-09-01        0.049   0.030   0.029    0.020       0.019     1
124 2010-07-07        0.052   0.043   0.031    0.021       0.009     1
151 2010-05-27        0.057   0.043   0.032    0.025       0.014     1
164 2010-05-10        0.073   0.064   0.043    0.030       0.009     1

[252 rows x 7 columns]

Question: Assume a risk-free asset returns 2% per year, and the standard deviation of returns of Dupont stock is 20%.  What is the Sharpe Ratio for Dupont stock for 2010? Give the answer to two digits to the right of the decimal place.

Question: What is the price x that maximizes weekly revenues?

Question: What is Beta for Dow Chemical? Give the answer rounded two digits to the right of the decimal place.

In [18]:
linregress(x=df["S&P500"], y=df["DowChemical"])

LinregressResult(slope=1.6577858774426197, intercept=0.00019595740659432903, rvalue=0.7850561872724646, pvalue=6.363169314348984e-54, stderr=0.08272677799986383)